### PR TITLE
refactor(credential): remove legacy single-key rotation path from Rotator

### PR DESCRIPTION
**Pull request by aaron-zeisler**
## Summary

Removes the legacy single-key rotation machinery from `internal/credential/rotator.go` now that all production credential flows go through `ReconcileCredentials` (#719 / SDK-2547 removed the last production callers of `RotateWithGrace`/`Rotate`).

[Jira: SDK-2603](https://launchdarkly.atlassian.net/browse/SDK-2603)

## Background

Prior to this project, the `Rotator` tracked credentials through a single-key rotation model: one primary SDK key, one deprecated-with-expiry slot (`deprecatedSdkKeys` / `deprecatedMobileKeys` maps), and `RotateWithGrace`/`Rotate` methods to move between them. The concurrent-keys work added a parallel `Reconcile` path with per-key expiry stored as data on accepted entries, and #719 wired both trusted-source handlers (RAC + offline) to `ReconcileCredentials`, leaving the legacy path with no production callers.

The `TEMPORARY` comment in `DeprecatedCredentials` pointed here as the cleanup step.

## Changes

The key conceptual shift: the old API was **imperative** ("rotate to this new key, here's the grace for the old one"), and the new API is **declarative** ("here is the complete desired accepted set; figure out the diff"). Most removed pieces don't have a 1:1 method replacement — they collapse into `Reconcile` and its helpers. The mapping below is old → new for everything this PR removes.

### Methods / entry points

| Removed | New analog | Notes |
|---|---|---|
| `Rotate(cred)` | `Reconcile(set, now)` | `Rotate` was just `RotateWithGrace(cred, nil)`. Caller now declares the full set instead of pushing one key at a time. |
| `RotateWithGrace(primary, grace)` | `Reconcile(set, now)` | The single declarative entry point; it internally sequences add → set-anchor → remove. |
| `updateSDKKey(sdkKey, grace)` | `reconcileSDKKeys(set, anchor, now)` | |
| `updateMobileKey(mobileKey, grace)` | `reconcileMobileKeys(set, now)` | |
| `updateEnvironmentID(envID)` (legacy) | `reconcileEnvironmentID(set)` | |
| `swapPrimaryKey(newKey)` | _(no dedicated method)_ | "Designate the anchor" is now just `desired[anchor] = nil` + `r.primarySdkKey = anchor` at the tail of `reconcileSDKKeys`. The map-consistency work it did is handled by the generic `reconcileAcceptedKeys`. |
| `immediatelyRevoke(key)` | _(no dedicated method)_ | Immediate revocation = *omit the key from the desired set*; the second pass of `reconcileAcceptedKeys` deletes any accepted key the set no longer wants and queues it as an expiration. |

### Types / data

| Removed | New analog | Notes |
|---|---|---|
| `GracePeriod` type, `NewGracePeriod()` | `acceptedKeyInfo.expiry *time.Time` (data on the accepted entry), supplied via `AcceptedSetBuilder.WithExpiringSDKKey(key, expiry)` / `WithExpiringMobileKey(...)` | A grace period is no longer a standalone object passed alongside a key — it's just an optional expiry field on the key's accepted entry. |
| `GracePeriod.Expired()` | inline time comparisons | On add: `!now.Before(*expiry)` filters in `reconcileSDKKeys`/`reconcileMobileKeys`. On the ticker: `now.After(*info.expiry)` in `StepTime`. |
| `deprecatedSdkKeys` / `deprecatedMobileKeys` maps | folded into `acceptedSDKKeys` / `acceptedMobileKeys` | A "deprecated" key is now just an accepted key with `info.expiry != nil`. One source of truth instead of two parallel maps. |

### Derived views / helpers

| Removed | New analog | Notes |
|---|---|---|
| `deprecatedCredentials()` (internal helper) | the inline loop in `DeprecatedCredentials()` | Now derived: accepted SDK keys where `info.expiry != nil && key != anchor`. |
| legacy grace-period loops in `StepTime` | the per-key-expiry loops in `StepTime` | These already coexisted; now they're the only loops. They read `info.expiry` instead of a separate map's value. |
| `PrimaryCredentials()` | `AllCredentials()` | Both returned the identical accepted set once the deprecated maps were gone, so they were consolidated to one. |

### Notes on imperfect alignment

- `expireSDKKey` / `expireMobileKey` were **not** removed. They're still the eviction routines `StepTime` calls; this PR only stripped the `delete(r.deprecatedSdkKeys, …)` lines out of their bodies.
- There is **no replacement** for the "stale deprecation message" handling. The old `updateSDKKey` had special logic for when LD re-sent a stale deprecation notice (the `previousExpiry, ok := r.deprecatedSdkKeys[grace.key]` branch). That whole class of problem disappears under reconcile — the caller always sends the *full current* desired set, so there is no stale-incremental-message state to reconcile against.

### Test changes

- `rotator_test.go`: all `RotateWithGrace`/`Rotate` tests removed; two new reconcile-path tests added: `TestReconcileAlreadyExpiredKeyIsIgnoredOnAdd` and `TestReconcileDeExpiryRestoresKey`.
- `env_context_impl_test.go`: `TestMobileKeyGraceExpiry` deleted (legacy path; superseded by the adjacent `TestMobileKeyReconcileExpiry`).
- `relay/filedata_actions_test.go`: remove dead `type foo struct{}`; fix `TestOfflineModeDeleteEnvironment` second `awaitEnvironment` call (was `Env1`, should be `Env2`); collapse `TestOfflineModeSDKKeyCanExpire` 3-iteration loop to a single run.
- `internal/credential/accepted_set_test.go`: remove no-op `_ = config.SDKKey(...)` assertion.

---

> [!NOTE]
> **Medium Risk**
> Changes authentication credential lifecycle in `Rotator` and what `GetCredentials` returns, but production paths already use `ReconcileCredentials`; risk is mainly regression in expiry/revocation edge cases covered by updated tests.
> 
> **Overview**
> Removes the legacy **imperative** credential rotation API (`Rotate`, `RotateWithGrace`, `GracePeriod`, and `deprecatedSdkKeys` / `deprecatedMobileKeys`) from `internal/credential/rotator.go`. Credential state is now maintained only through **`Reconcile`** and the `acceptedSDKKeys` / `acceptedMobileKeys` maps, where “deprecated” keys are simply accepted entries with a non-nil `expiry`.
> 
> **`PrimaryCredentials()` is removed** and **`AllCredentials()`** becomes the single view of every accepted key (including those still valid until `StepTime` evicts them). **`DeprecatedCredentials()`** no longer merges legacy grace maps—it only lists non-anchor SDK keys with a future expiry. **`StepTime`** and **`reconcileAcceptedKeys`** drop the parallel deprecated-map logic; expiry is enforced solely from `acceptedKeyInfo.expiry`.
> 
> **`env_context_impl.GetCredentials`** now delegates to **`AllCredentials()`** instead of `PrimaryCredentials()`.
> 
> Tests shed all `Rotate`/`RotateWithGrace` coverage and add reconcile cases (`TestReconcileAlreadyExpiredKeyIsIgnoredOnAdd`, `TestReconcileDeExpiryRestoresKey`); **`TestMobileKeyGraceExpiry`** is removed. **`relay/filedata_actions_test.go`** fixes a wrong second `awaitEnvironment` env ID and simplifies the SDK key expiry test loop.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 9d8fc83d997d669b682b9883db4eebb9406f2984. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/credential/rotator.go [hunk 1]
```diff
@@ -19,24 +19,19 @@ type Rotator struct {
...
-	// by credential value with the associated expiry time. StepTime walks this map and emits
-	// an expiration once a key's grace period passes, mirroring deprecatedSdkKeys.
-	deprecatedMobileKeys map[config.MobileKey]time.Time
-
 	// There is only one environment ID active at a given time, and it won't actually be rotated. The mechanism is
 	// here to allow setting it in a deferred manner.
 	primaryEnvironmentID config.EnvironmentID
 
-	// There can be multiple SDK keys active at a given time, but only one is primary.
+	// There can be multiple SDK keys active at a given time, but only one is primary (the anchor).
```
Comment (aaron-zeisler) [inline internal/credential/rotator.go]: Don't change this line - the change is unnecessary, and it'll create a merge conflict with PR #722 .

File: internal/credential/rotator.go [hunk 2]
```diff
@@ -177,230 +147,41 @@ func (r *Rotator) DeprecatedCredentials() []SDKCredential {
 	return out
 }
 
-// AllCredentials returns the primary and deprecated credentials as one list.
+// AllCredentials returns all accepted credentials. It is equivalent to PrimaryCredentials; both
+// methods return the full accepted set (accepted keys, including those carrying a future expiry,
+// together with the environment ID).
```
Comment (aaron-zeisler) [inline internal/credential/rotator.go]: It seems unnecessary to keep both PrimaryCredentials() and AllCredentials(). Let's just keep just one - analyze the code to understand the reason why they both exist and show me the call sites.

File: internal/credential/rotator.go [hunk 3]
```diff
@@ -177,230 +147,41 @@ func (r *Rotator) DeprecatedCredentials() []SDKCredential {
...
-// It enforces expiry from both expiry mechanisms, for both SDK and mobile keys:
-//   - The legacy grace-period maps (deprecatedSdkKeys / deprecatedMobileKeys), populated by the
-//     RotateWithGrace path, where the expiry lives in the map value.
-//   - The reconcile path, where per-key expiry is stored as data on the accepted entry
-//     (acceptedKeyInfo.expiry); a nil expiry means the key is permanent and is never expired here.
+// It enforces per-key expiry for both SDK and mobile keys: expiry is stored as data on the accepted
+// entry (acceptedKeyInfo.expiry); a nil expiry means the key is permanent and is never expired here.
 //
-// Expiry is strict (now strictly after the expiry timestamp), consistent across all four loops.
+// Expiry is strict (now strictly after the expiry timestamp).
```
Comment (aaron-zeisler) [inline internal/credential/rotator.go]: Change this to read "Expiry happens strictly after a key's expiry timestamp". The wording "now strictly after ..." refers to the change in this project, but we want the comments to reflect the current state of the code without the need for understanding its history. If there are other similar comments that express or explain the code's history, please update them to reflect the code's current state.

**Review by keelerm84 — APPROVED** [review]
